lstchain/reco/utils.py

```diff
@@ -649,12 +655,37 @@ def get_effective_time(events):
     t_eff: float
     t_elapsed: float
     """
-    
+
+    # time differences between the events in the table (which in general are
+    # NOT all triggered events):
+    time_diff = np.diff(events.dragon_time)
+    # elapsed time: sum of those time differences, excluding large ones which
+    # might indicate the DAQ was stopped (e.g. if the table contains more
+    # than one run). We set 0.1 s as limit to decide a "break" occurred:
+    t_elapsed = np.sum(time_diff[time_diff<0.1])
+
+    # Get delta_t, the time elapsed since the previous triggered event.
+    # We exclude the first event, for which delta_t might be set to 0 if it is
+    # the first even in a file.
+    # We want this to calculate the actual trigger rate, on which the dead
+    # time depends.
     delta_t = events.delta_t[1:]
```

Comment:
do you start on the second index (`events.delta_t[1:]`) because the first one is set to 0?
if for whatever reason we calculate `delta_t` per subrun and then calculate the effective time per run, then we will have several zeros in the `delta_t` table for the first run of every subrun. Although it would be better not to have this behaviour and calculate the `delta_t` table per run, I would add some check like:
`delta_t = events.delta_t[events.delta_t > 0.]`

Reply:
> why is this more general for this particular case?
Not sure what you mean, but to obtain the elapsed time we need the intervals between the events present in the file, that's all.

Reply:
> do you start on the second index (`events.delta_t[1:]`) because the first one is set to 0?

yes

> if for whatever reason we calculate `delta_t` per subrun and then calculate the effective time per run, then we will have several zeros in the `delta_t` table for the first run of every subrun. Although it would be better not to have this behaviour and calculate the `delta_t` table per run, I would add some check like:
> `delta_t = events.delta_t[events.delta_t > 0.]`

Good point, thanks!, will do!

Reply:
but if for example we add `delta_t` to dl2 files that come from dl1 subruns, the first event of every subrun will always have `delta_t=0`.
If we then merge all these dl2 subruns and calculate the effective time afterwards, the `np.amin(delta_t)` will be 0

Reply:
because by skipping the first event of the `events` data frame (`events.delta_t[1:]`) you are skipping the first event of the file, but the file could have been merged after the `delta_t` have been calculated

Reply:
Thanks, all the suggestions implemented.
Besides, I realized that in the test the true livetime for comparison could be better calculated. Now the accuracy of the method is ~0.02%, so I also reduced the tolerance in the test.

Reply:
> > why is this more general for this particular case?
> 
> Not sure what you mean, but to obtain the elapsed time we need the intervals between the events present in the file, that's all.

I think this was copied for my earlier comment, by mistake